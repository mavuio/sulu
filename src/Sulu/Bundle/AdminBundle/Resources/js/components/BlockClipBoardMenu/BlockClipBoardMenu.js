// @flow
import React from 'react';
import blockclipboardmenuStyles from './blockclipboardmenu.scss';
// import BlockClipboardStore from '../../stores/BlockClipboardStore';
import userStore from '../../stores/userStore';
import { toJS} from 'mobx';

const USER_SETTING_CLIPBOARD = 'sulu_page.block_clipboard';

type Props = {|
    value: Object,
|};

export default class BlockClipBoardMenu extends React.PureComponent<Props> {

    onCopyClick(value) {
        let item=toJS(value);
        this.setClipboardItems([item]);
        console.log('#log 8534 copy',item);
    }

    onPasteClick(position) {
        const {onPasteBlocks} = this.props;
        let items=this.getClipboardItems().map( item => toJS(item));
        console.log('#log 7987 paste',items);
        onPasteBlocks(items);
    }

    setClipboardItems(items) {
        userStore.setPersistentSetting(USER_SETTING_CLIPBOARD, items);
    }

    getClipboardItems() {

        return userStore.getPersistentSetting(USER_SETTING_CLIPBOARD);
    }

    copy_icon=(
        <svg class="w-6 h-6" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M8 5H6a2 2 0 00-2 2v12a2 2 0 002 2h10a2 2 0 002-2v-1M8 5a2 2 0 002 2h2a2 2 0 002-2M8 5a2 2 0 012-2h2a2 2 0 012 2m0 0h2a2 2 0 012 2v3m2 4H10m0 0l3-3m-3 3l3 3"></path></svg>
    )
    cut_icon=(
        <svg class="w-6 h-6" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M14.121 14.121L19 19m-7-7l7-7m-7 7l-2.879 2.879M12 12L9.121 9.121m0 5.758a3 3 0 10-4.243 4.243 3 3 0 004.243-4.243zm0-5.758a3 3 0 10-4.243-4.243 3 3 0 004.243 4.243z"></path></svg>
    )
    duplicate_icon=(
        <svg class="w-6 h-6" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M8 16H6a2 2 0 01-2-2V6a2 2 0 012-2h8a2 2 0 012 2v2m-6 12h8a2 2 0 002-2v-8a2 2 0 00-2-2h-8a2 2 0 00-2 2v8a2 2 0 002 2z"></path></svg>
    )

    multiselect_icon=(<svg class="w-6 h-6" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5H7a2 2 0 00-2 2v12a2 2 0 002 2h10a2 2 0 002-2V7a2 2 0 00-2-2h-2M9 5a2 2 0 002 2h2a2 2 0 002-2M9 5a2 2 0 012-2h2a2 2 0 012 2m-6 9l2 2 4-4"></path></svg>)

    render() {
        const {value} = this.props;

        return (
            <ul className={blockclipboardmenuStyles.blockclipboardmenu}>
                    <li ><span>{this.multiselect_icon}</span>select multiple</li>
                    <li  onClick={() => this.onCopyClick(value)}><span>{this.copy_icon}</span>copy to clipboard</li>
                    <li  onClick={() => this.onCopyClick(value)}><span>{this.cut_icon}</span>move to clipboard ("cut")</li>
                    <li  onClick={() => this.onCopyClick(value)}><span>{this.duplicate_icon}</span>duplicate</li>

                    {/* <li onClick={ () => this.onPasteClick('before')} >paste n elementfrom clipboard BEFORE this element</li>
                    <li onClick={ () => this.onPasteClick('after')}>paste n element(s) from clipboard AFTER this element</li> */}
                    {/* <li >create new element BEFORE this element</li> */}
                    {/* <li >create new element AFTER this element</li> */}
            </ul>);
    }
}
