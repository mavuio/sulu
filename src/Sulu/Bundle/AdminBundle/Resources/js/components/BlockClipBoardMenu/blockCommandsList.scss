@import "../../containers/Application/colors.scss";

$blockCommandsListBackgroundColor: $alto;
$blockCommandsListTransparentBackgroundColor: $white;
$blockCommandsListBoxShadowColor: rgba($dustyGray, 0.5);
$blockCommandsListTransparentBoxShadowColor: rgba($alto, 0.5);

.block-commands-list {
    list-style: none;
    margin: 0;
    padding: 0;
    white-space: nowrap;
    max-height: 70vh;
    min-width: 10rem;
    overflow: auto;
    display: flex;
    flex-direction: column;
    padding: 1rem;
    background-color: $blockCommandsListBackgroundColor;
    box-shadow: 2px 6px 12px 0 $blockCommandsListBoxShadowColor;
    > li {
        margin: 0.2rem 0;
    }
}
