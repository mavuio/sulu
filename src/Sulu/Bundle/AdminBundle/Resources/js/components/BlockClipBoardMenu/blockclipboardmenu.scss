@import "../../containers/Application/colors.scss";

$blockCommandsListBackgroundColor: $alto;
$blockCommandsListTransparentBackgroundColor: $white;
$blockCommandsListBoxShadowColor: rgba($dustyGray, 0.5);
$blockCommandsListTransparentBoxShadowColor: rgba($alto, 0.5);

.blockclipboardmenu {
    list-style: none;
    margin: 0;
    white-space: nowrap;
    max-height: 70vh;
    min-width: 10rem;
    overflow: auto;
    display: flex;
    padding: 1rem 0;
    flex-direction: column;
    background-color: $blockCommandsListBackgroundColor;
    box-shadow: 2px 6px 12px 0 $blockCommandsListBoxShadowColor;
    span {
        width: 2em;
        margin-right: 0.5em;
        vertical-align: middle;
        display: inline-block;
    }

    > li {
        &:hover {
            background-color: #bbb;
        }

        display: block;
        margin: 0;
        color: #222;
        padding: 0.5em 0.9em;
        text-decoration: none;
    }
}
