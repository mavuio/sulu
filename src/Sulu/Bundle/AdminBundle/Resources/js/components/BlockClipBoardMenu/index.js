// @flow
import BlockClipBoardMenu from './BlockClipBoardMenu';

export default BlockClipBoardMenu;
