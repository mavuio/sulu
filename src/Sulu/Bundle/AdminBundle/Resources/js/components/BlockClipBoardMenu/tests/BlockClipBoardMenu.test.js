// @flow
import React from 'react';
import {render} from 'enzyme';
import BlockClipBoardMenu from '../BlockClipBoardMenu';

test('Render a blockclipboardmenu', () => {
    expect(render(<BlockClipBoardMenu>Hello world</BlockClipBoardMenu>)).toMatchSnapshot();
});
