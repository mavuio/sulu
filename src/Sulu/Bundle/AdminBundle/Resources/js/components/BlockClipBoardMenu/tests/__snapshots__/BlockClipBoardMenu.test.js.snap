// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a blockclipboardmenu 1`] = `
<div
  class="blockclipboardmenu"
>
  Hello world
</div>
`;
